selinux: update to latest release

PolynomialDivision opened the pull request:
Merge PRs into one to test it against CI.

misaka36 committed:
libselinux: update to 3.5

Switch from libpcre to libpcre2. While working on it remove the double
defined HOST_BUILD_DEPENDS section.

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>
[depend on libpcre2]
Signed-off-by: Nick Hainke <vincent@systemli.org>

misaka36 committed:
libsemanage: update to 3.5

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>

misaka36 committed:
libsepol: update to 3.5

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>

misaka36 committed:
checkpolicy: update to 3.5

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>

misaka36 committed:
policycoreutils: update to 3.5

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>

misaka36 committed:
secilc: update to 3.5

Release Notes:
https://github.com/SELinuxProject/selinux/releases/download/3.4/RELEASE-3.4.txt
https://github.com/SELinuxProject/selinux/releases/download/3.5/RELEASE-3.5.txt

Signed-off-by: Linhui Liu <liulinhui36@gmail.com>

PolynomialDivision committed:
pcre2: import pcre2 from packages feed

pcre2 is needed by newer selinux versions, so it needs to be in the base
repository.

Signed-off-by: Nick Hainke <vincent@systemli.org>

robimarko commented:
Its weird to see it fail, compiles locally just fine

lovehackintosh commented on package/libs/libselinux/Makefile:
```diff
@@ -6,12 +6,12 @@
 PKG_SOURCE_URL:=https://github.com/SELinuxProject/selinux/releases/download/$(PKG_VERSION)
-PKG_HASH:=acfdee27633d2496508c28727c3d41d3748076f66d42fccde2e6b9f3463a7057
+PKG_HASH:=9a3a3705ac13a2ccca2de6d652b6356fead10f36fb33115c185c5ccdf29eec19
 HOST_BUILD_DEPENDS:=libsepol/host pcre/host
```
Why this package defined HOST_BUILD_DEPENDS twice?
And pcre/host should be pcre2/host

PolynomialDivision commented:
> Though there is a bunch of packages depending on `pcre` and not `pcre2`

But they are only located in the openwrt/packages?

Yeah, it's weird. Everything builds also fine on my system. I will now test again with docker container. Let's see why it fails.

misaka36 reviewed (commented):
Commit message has a typo error, doulbe should be double

PolynomialDivision commented:
Builds now fine! :) So we can move pcre to openwrt/packages and than merge this here.

ynezz commented:
>  We can't move pcre to openwrt/packages because some packages of telephony are using it.

Yes, you can do that and should do that. If you for example look at asterisk dependencies, you'll find out, that you won't be able to built it without packages feed anyway, its not free standing feed.

PolynomialDivision committed:
pcre: move package to packages feed

With the update of selinux no package depends anymore on pcre in the
base repository. Move it to packages feed.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
merged to master. @lovehackintosh thanks! :)

PolynomialDivision commented:
@lovehackintosh I think you forgot to update selinux-policy?

lovehackintosh commented:
> @lovehackintosh I think you forgot to update selinux-policy?

#12743
